ctapipe/tools/stage1.py

```diff
@@ -0,0 +1,868 @@
+    ).tag(config=True)
+
+    compression_type = CaselessStrEnum(
+        values=["blosc:zstd", "zlib"],
+        help="compressor algorithm to use. ",
+        default_value="zlib",
+    ).tag(config=True)
+
+    image_extractor_type = enum_trait(
+        base_class=ImageExtractor,
+        default="NeighborPeakWindowSum",
+        help_str="Method to use to turn a waveform into a single charge value",
+    ).tag(config=True)
+
+    gain_selector_type = enum_trait(
+        base_class=GainSelector, default="ThresholdGainSelector"
+    ).tag(config=True)
+
+    image_cleaner_type = enum_trait(
+        base_class=ImageCleaner, default="TailcutsImageCleaner"
+    )
+
+    write_index_tables = Bool(
+        help=(
+            "Generate PyTables index datasets for all tables that contain an "
+            "event_id or tel_id. These speed up in-kernal pytables operations,"
+            "but add some overhead to the file. They can also be generated "
+            "and attached after the file is written "
+        ),
+        default_value=False,
+    ).tag(config=True)
+
+    overwrite = Bool(help="overwrite output file if it exists").tag(config=True)
+    progress_bar = Bool(help="show progress bar during processing").tag(config=True)
+
+    aliases = {
+        "input": "EventSource.input_url",
+        "output": "Stage1ProcessorTool.output_filename",
+        "allowed-tels": "EventSource.allowed_tels",
+        "max-events": "EventSource.max_events",
+        "image-extractor-type": "Stage1ProcessorTool.image_extractor_type",
+        "gain-selector-type": "Stage1ProcessorTool.gain_selector_type",
+        "image-cleaner-type": "Stage1ProcessorTool.image_cleaner_type",
+    }
+
+    flags = {
+        "write-images": (
+            {"Stage1ProcessorTool": {"write_images": True}},
+            "store DL1/Event/Telescope images in output",
+        ),
+        "write-parameters": (
+            {"Stage1ProcessorTool": {"write_images": True}},
+            "store DL1/Event/Telescope parameters in output",
+        ),
+        "write-index-tables": (
+            {"Stage1ProcessorTool": {"write_index_tables": True}},
+            "generate PyTables index tables for the parameter and image datasets",
+        ),
+        "overwrite": (
+            {"Stage1ProcessorTool": {"overwrite": True}},
+            "Overwrite output file if it exists",
+        ),
+        "progress": (
+            {"Stage1ProcessorTool": {"progress_bar": True}},
+            "show a progress bar during event processing",
+        ),
+    }
+
+    classes = List(
+        [EventSource, CameraCalibrator, ImageSelector]
+        + classes_with_traits(ImageCleaner)
+        + classes_with_traits(ImageExtractor)
+        + classes_with_traits(GainSelector)
+    )
+
+    def setup(self):
+
+        # prepare output path:
+
+        output_path = Path(expandvars(self.output_filename)).expanduser()
+        if output_path.exists() and self.overwrite:
```

Comment:
what about if `output_path.exists() and not self.overwrite`

Reply:
Karl answered this above:

> You get a message from the pytables library - it will not overwrite by default, and can only add datasets, not modify existing ones.

Reply:
we could check and throw a better error message though (the message from pytables is not always clear)

Reply:
Better explicitly raise an error here

Reply:
done